### PR TITLE
PWA-1564: [bug]: Guest checkout, France region validation failed

### LINKED ISSUE #3057: [bug]: Guest checkout, France region validation failed
**Describe the bug**

As guest, there is an error about address region (with country = France) when customer wants to proceed to shipping step.

**To reproduce**

1. Go to the checkout as guest with a valid cart
1. Set the address country with 'France'
1. Set the region with 'Ain' (the first one)
1. Fill all the required fields as usual
1. Click on "Continue to Shipping Method" button
1. Error "Invalid value of "1" provided for the regionId field." is displayed

**Expected behavior**

The address should be valid.

**Screenshots**

Capture

**Possible solutions**

France region code are mostly numeric, I think that's the issue. See \Magento\Customer\Model\Address\AbstractAddress::getRegionCode

**Please complete the following device information:**
 - Magento 2.4.2
 - PWA Studio 9.0.1

**Acceptance Criteria**
1. Root cause analysis is performed and actual underlying issue is fixed. In other words, we know that France doesn't work but do other countries have the same problem?
1. Users can successfully submit addresses in France (and all countries affected by this bug)

### PR DESCRIPTION
**Pull request by anthoula**
## Description

Some region codes are inconsistent across countries. Update region field to use region_id instead of region_code to help fix inconsistencies in GraphQL response.

## Related Issue

Closes #3057

## Related Pull Requests
* https://github.com/magento-commerce/pwa-tests/pull/13

### Verification Stakeholders
@dpatil-magento 

### Verification Steps

1. Go to the checkout as guest with a valid cart
2. Set the address country with 'France'
3. Set the region with 'Ain' (the first one)
4. Fill all the required fields as usual
5. Click on "Continue to Shipping Method" button
6. Select Shipping Method
7. Click on "Continue to Payment Information" button
8. Enter required payment information
9. Ensure a France billing address is selected
9. Click on "Review Order"
10. Click on "Place Order"
11. An order is successfully placed

Additional scenarios outlined in [PWA-1564](https://jira.corp.magento.com/browse/PWA-1564)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/creditCard.gql.js [hunk 1]
```diff
@@ -63,6 +65,7 @@ export const GET_SHIPPING_ADDRESS = gql`
                 city
                 region {
                     code
+                    region_id
```
Comment (supernova-at) [inline packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/creditCard.gql.js:68]: Do we not need `label` for this one?

**Review by supernova-at — DISMISSED** [review]
Code looks good, verification steps passed ✅

**Review by supernova-at — APPROVED** [review]

**Comment by supernova-at** [comment]
During code review / demo, @anthoula pointed out that we should add editing / initial value testing to the verification / QA steps

**Comment by tjwiebell** [comment]
@anthoula 
All manual test cases pass and original bug is no longer reproducible. This did unfortunately break an MFTF test, `CartPageRegisteredUserCheckout`. 

Screen Shot 2021-04-23 at 4 09 40 PM

I debugged a little, and it appears this selector is no longer unique after your change, `[class*='select-input-'][name*='region[region_id]']`. I'm not sure if Dev blocked merging based on failures and would just fix them in post, but happy to resolve this failure for you if you're not ramped up on the test suite yet. Ping me on Slack and we'll decide how to proceed.

**Pull request merged**